content/content-tool/tool/src/java/org/sakaiproject/content/tool/FilePickerAction.java

```diff
@@ -1144,7 +1145,14 @@ public void doAttachitem(RunData data)
 
 		//toolSession.setAttribute(STATE_LIST_SELECTIONS, new TreeSet());
 
-		String itemId = params.getString("itemId");
+		String itemId = null;
+		try {
+		    itemId = URLDecoder.decode(params.getString("itemId"), "UTF-8");
+		} catch (java.io.UnsupportedEncodingException e) {
+		    // should be impossible. use original, with warning
+		    logger.warn("UTF-8 unsupported???");
```

Comment:
I understand. But the compiler requires a catch (or declaring the method to return an error). I think it’s better to log something if the catch occurs even if it should be impossible. I could also use the one-argument decode, but I’d rather not depend upon what the default setting for the VM is.

> On Mar 11, 2016, at 10:16 AM, Matthew Buckett <notifications@github.com> wrote:
> 
> In content/content-tool/tool/src/java/org/sakaiproject/content/tool/FilePickerAction.java <https://github.com/sakaiproject/sakai/pull/1898#discussion_r55841714>:
> 
> > @@ -1144,7 +1145,14 @@ public void doAttachitem(RunData data)
> >  
> >  		//toolSession.setAttribute(STATE_LIST_SELECTIONS, new TreeSet());
> >  
> > -		String itemId = params.getString("itemId");
> > +		String itemId = null;
> > +		try {
> > +		    itemId = URLDecoder.decode(params.getString("itemId"), "UTF-8");
> > +		} catch (java.io.UnsupportedEncodingException e) {
> > +		    // should be impossible. use original, with warning
> > +		    logger.warn("UTF-8 unsupported???");
> UTF-8 is guaranteed to be available. https://docs.oracle.com/javase/7/docs/api/java/nio/charset/StandardCharsets.html <https://docs.oracle.com/javase/7/docs/api/java/nio/charset/StandardCharsets.html>
> It's just a shame there's not a URLDecoder method that accepts a Charset.
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/sakaiproject/sakai/pull/1898/files#r55841714>.
>